subsys/net/ip/tcp2.c

```diff
@@ -924,15 +1073,19 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 			break;
 		}
 
+		update_acked_data(conn, th);
+
 		if (len) {
 			if (th_seq(th) == conn->ack) {
 				tcp_data_get(conn, pkt);
 				conn_ack(conn, + len);
 				tcp_out(conn, ACK);
-			} else if (th_seq(th) < conn->ack) {
+			} else if (net_tcp_seq_greater(conn->ack,
```

Comment:
Shall we just use simple '>' here?

Reply:
Wouldn't it fail if the numbers have wrapped?